### PR TITLE
Django 4.0 Support

### PR DESCRIPTION
**Pull request by smithdc1**
Django 4.0 is coming and the test suite has a number of failures. 

This is intended to fix the first couple of issues, which leaves another 6 failures at the time of writing.

### CODE REVIEW

**Comment by smithdc1** [comment]
Ok, the test suite now passes with Django 4.0. I've started to update some of the docs `setup.py` etc as well.

**Comment by smithdc1** [comment]
This will mean that Django 4.0 will require the changes here (likely the next version). I think this will be the first release in a while which has caused crispy-forms to _need_ new release. 

We should call it out clearly that Django 4.0+ needs this version or higher. I think this is just a docs issue? 🤔

**Comment by smithdc1** [comment]
Phew, the tests still pass 😉

File: setup.py [hunk 1]
```diff
@@ -31,13 +31,15 @@
         "Framework :: Django :: 2.2",
         "Framework :: Django :: 3.1",
         "Framework :: Django :: 3.2",
+        "Framework :: Django :: 4.0",
```
Comment (carltongibson) [inline setup.py:34]: This doesn't **yet** exist. (But will soonish...)

**Comment by smithdc1** [comment]
@carltongibson -- I think this is all ready, any comments? 

We can push a new release with the classifier as well now?

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       matrix:
         python: [3.6, 3.7, 3.8, 3.9]
-        django: [2.2.*, 3.0.*, 3.1.*]
+        django: [2.2.*, 3.1.*, 3.2.*]
```
Comment (carltongibson) [inline .github/workflows/publish.yml:14]: 4.0a1?

File: setup.py (hunk 1 above)
Comment (carltongibson) [reply setup.py:34]: Yes, this is live now. 💃

**Review by carltongibson — APPROVED** [review]
This looks good 👍 Thanks @smithdc1

**Comment by carltongibson** [comment]
Good stuff.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (smithdc1) [reply .github/workflows/publish.yml:14]: Should probably replace this with the tox version like the workflow which runs on PRs. 

🕵️

Will investigate.

Comment (smithdc1) [reply .github/workflows/publish.yml]: Ok, I've updated this. I guess there's only one way to test this! 😬

**Pull request merged**